### DISCUSSION TITLE
Question about Web dev setup

### DISCUSSION THREAD

**Discussion by dpetka2001** [LazyVim/LazyVim#434] [Q&A] [answered]
So, I recently started a seminar on designing web pages. We are using quite a lot of `<script>` tags inside a .html file and I don't get any completion from LSP for javascript(since it's not even attached). I use only `html` LSP server. When i write in an external .js file `typescript` LSP works as intended. What would be the best way to achieve javascript completion from the LSP server inside html files. Just enable `typescript` for .html files as well? Or maybe some other recommendation?

**Comment by Bruno-Luna** [comment]
before the closing `</body>`  tag, insert the line with the index of the script. Example below:

![image](https://user-images.githubusercontent.com/80595156/225735130-3386bbdc-3e65-4042-ad06-f6341a55c0fa.png)
![image](https://user-images.githubusercontent.com/80595156/225736299-8aae6f6d-2df6-4610-926a-ac0a3cb67d0a.png)

**Reply by dpetka2001** [reply]
I'm not talking about this. I'm talking about not getting embedded Javascript Lsp completion in html files. Whereas in pure .js files typescript Lsp works just fine.

**Reply by dpetka2001** [reply]
I saw in a nvim-lspconfig issue that `html` lsp server does not seem to support this feature. So I guess I'll just go with seperate .js files henceforward and make use of `typescript` server.

**Comment by folke** [comment, marked as answer]
As far as I know embedded languages are **not** supported by Neovim, so if the lsp server does not support it themselves, another lsp server won't automatically connect to that region.

**Comment by dpetka2001** [comment]
For anyone else wondering about the same thing. I got a reply in Reddit about a plugin called [`inline_edit`](https://github.com/AndrewRadev/inline_edit.vim/).  It opens the tag you want to edit in a proxy buffer with correct LSP attachment and you get `typescript` LSP autocompletion. It bridges the gap between lack of embedded language support from Neovim. For my use case it is exactly what I needed. I will still use external .js files as more correct practice, but I'm really glad I found this plugin.
